File: internal/codemap/testdata/typescript/simple.ts
import { readFileSync } from 'fs';
import { helper } from './local';
import express from 'express';

export function greet(name: string): string {
    return `Hello, ${name}!`;
}

export class MyClass {
    private value: number;

    constructor(value: number) {
        this.value = value;
    }

    getValue(): number {
        return this.value;
    }
}

export interface Handler {
    handle(input: string): void;
}

export type Config = {
    host: string;
    port: number;
};

export enum Status {
    Active = 'active',
    Inactive = 'inactive',
}

File: internal/codemap/testdata/rust/simple.rs
use std::io;
use crate::util;
use serde::Serialize;

pub struct Config {
    pub host: String,
    pub port: u16,
}

pub enum Status {
    Active,
    Inactive,
}

pub trait Handler {
    fn handle(&self, input: &str) -> String;
}

impl Config {
    pub fn new(host: String, port: u16) -> Self {
        Config { host, port }
    }

    pub fn address(&self) -> String {
        format!("{}:{}", self.host, self.port)
    }
}

pub fn hello() -> String {
    String::from("hello")
}

fn private_helper() -> bool {
    true
}
